apps/storybook/.storybook/main.ts

```diff
@@ -0,0 +1,38 @@
+import type { StorybookConfig } from "@storybook/react-vite";
+
+import { join, dirname } from "path";
+
+/**
+ * This function is used to resolve the absolute path of a package.
+ * It is needed in projects that use Yarn PnP or are set up within a monorepo.
+ */
+function getAbsolutePath(value: string): any {
+  return dirname(require.resolve(join(value, "package.json")));
```

Comment:
I am not sure how it even works 🤔 
it's "type": "module" in the package.json
and this is cleary CJS
most likely you need to rename the file to `main.cts`

Reply:
It's autogenerated by storybook so let's keep it like this